validator/main.py

```diff
@@ -108,16 +108,28 @@ def __init__(
             self._model_threshold = model_threshold
 
         self.set_callable(llm_callable)
+        self.classifier = pipeline(
+            "zero-shot-classification",
+            model=self._model,
+            device=self._device,
+            hypothesis_template="This example has to do with topic {}.",
+            multi_label=True,
+        )
 
     def get_topic_ensemble(
         self, text: str, candidate_topics: List[str]
     ) -> ValidationResult:
-        topic, confidence = self.get_topic_zero_shot(text, candidate_topics)
-
-        if confidence > self._model_threshold:
-            return self.verify_topic(topic)
-        else:
-            return self.get_topic_llm(text, candidate_topics)
+        topics, scores = self.get_topic_zero_shot(text, candidate_topics)
```

Comment:
It looks like the way we're doing ensembling here is by doing a first pass by the zero shot model and the final decisions made by the LLM. IMO, a more robust (and cosistent with the SensitiveTopics validator) approach would be to do get 'votes' by the zero-shot classifier and the LLM, and if at least one is confident then we determine that that topic is present.

Regardless, the approach used here and in SensitiveTopics should be consistent.